CMakeLists.txt

```diff
@@ -431,22 +431,26 @@ include(ClangTidy)
 enable_clang_tidy(
     CHECKS
         *
-        -abseil-string-find-startswith
+        -abseil-*
         -android-cloexec-fopen
         # Yea we shouldn't be using rand()
         -cert-msc30-c
         -bugprone-exception-escape
+        -bugprone-macro-parentheses
         -cert-env33-c
         -cert-msc32-c
         -cert-msc50-cpp
         -cert-msc51-cpp
+        -cert-dcl37-c
+        -cert-dcl51-cpp
         -clang-analyzer-alpha.core.CastToStruct
         -clang-analyzer-optin.performance.Padding
         -clang-diagnostic-deprecated-declarations
         -clang-diagnostic-extern-c-compat
         -clang-diagnostic-unused-command-line-argument
         -cppcoreguidelines-avoid-c-arrays
         -cppcoreguidelines-avoid-magic-numbers
+        -cppcoreguidelines-avoid-non-const-global-variables
```

Comment:
This seems like an important check that we should not disable. Why is this necessary?

Reply:
Because I thought we were using a lot of global variables like " extern bool"
https://github.com/ROCmSoftwarePlatform/MIOpen/blob/f8f7e325cd2a9c2e0a7e3944f226c8b800b87968/src/include/miopen/logger.hpp#L202
https://github.com/ROCmSoftwarePlatform/MIOpen/blob/f8f7e325cd2a9c2e0a7e3944f226c8b800b87968/src/include/miopen/solver.hpp#L53
https://github.com/ROCmSoftwarePlatform/MIOpen/blob/f8f7e325cd2a9c2e0a7e3944f226c8b800b87968/src/include/miopen/find_controls.hpp#L41
https://github.com/ROCmSoftwarePlatform/MIOpen/blob/f8f7e325cd2a9c2e0a7e3944f226c8b800b87968/src/include/miopen/find_db.hpp#L64

https://github.com/ROCmSoftwarePlatform/MIOpen/blob/f8f7e325cd2a9c2e0a7e3944f226c8b800b87968/src/solver/conv_direct_naive_conv.cpp#L38

but there were only 5 of them

Reply:
I'll just add them to the exceptions

Reply:
Can we and `NOLINT` to those lines instead?

Also, `AlwaysEnableConvDirectNaive` can be removed as its not used.

Reply:
Yes, by exception I meant NOLINT

Reply:
> Also, AlwaysEnableConvDirectNaive can be removed as its not used.

It is used.